README.md

```diff
@@ -1,5 +1,130 @@
+
+*   _cmake 3.9+_
+*   C++11 compliant compiler, one of:
+    *   _gcc 5.3+, 6.3+, 7.3+, 8.1+_
+    *   _clang 3.9+_
+    *   _Apple LLVM 8.0+_ (__TODO__: verify)
+
+The Ginkgo CUDA module has the following __additional__ requirements:
+
+*   _CUDA 9.0+_
+*   Any host compiler restrictions your version of CUDA may impose also apply
+    here. For the newest CUDA version, this information can be found in the
+    [CUDA installation guide for Linux](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html)
+    or [CUDA installation guide for Mac Os X](https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html)
+
+In addition, if you want to contribute code to Ginkgo, you will also need the
+following:
+
+*   _clang-format 5.0.1+_ (ships as part of _clang_)
+
+### Windows
+
+Windows is currently not supported, but we are working on porting the library
+there. If you are interested in helping us with this effort, feel free to
+contact one of the developers. (The library itself doesn't use any non-standard
+C++ features, so most of the effort here is in modifying the build system.)
+
+__NOTE:__ Some restrictions will also apply on the version of C and C++ standard
+libraries installed on the system. This needs further investigation.
+
+Quick Install
+------------
+
+### Building Ginkgo
+
+To build Ginkgo, you can use the standard CMake procedure. 
+
+```sh
+mkdir build; cd build
+cmake -G "Unix Makefiles" .. && make
+```
+
+By default, `GINKGO_BUILD_REFERENCE` is enabled. You should be able to run examples with this
+executor. You would need to explicitly compile with the OpenMP and CUDA modules enabled
+to run with these executors. Please refer to the [Installation page](./INSTALL.md).
+
+After the installation, CMake can find ginkgo with `find_package(Ginkgo)`.
+An example can be found in the [`test_install`](test_install/CMakeLists.txt).
+
+### Ginkgo Examples
+
+Various examples are available for you to understand and play with Ginkgo within the `examples/` directory. They can be compiled by passing the `-DGINKGO_BUILD_EXAMPLES=ON` to the cmake command. Documentation for the examples is available within the `doc/` folder in each of the example directory and a commented code with explanations can found in the [online documentation](https://ginkgo-project.github.io/ginkgo/doc/master/Examples.html).
+
+### Ginkgo Testing
+
+Ginkgo does comprehensive unit tests using Google Tests. These tests are enabled by default and can be disabled if necessary by passing the `-DGINKGO_BUILD_TESTS=NO` to the cmake command. More details about running tests can be found in the [TESTING.md page](./TESTING.md).
+
+### Running the benchmarks
+
+A unique feature of Ginkgo is the ability to run benchmarks and view your results
+with the help of the [Ginkgo Performance Explorer (GPE)](https://ginkgo-project.github.io/gpe/). 
+
+More details about this can be found in the [documentation](https://ginkgo-project.github.io/ginkgo/doc/master/)
```

Comment:
This link is also not working at the moment: https://ginkgo-project.github.io/ginkgo/doc/master/

Reply:
Documentation was not generated for master yet. It will once this is merged and the step `gh-pages` is ran.